articles/underspecification/index.md

```diff
@@ -0,0 +1,105 @@
+
+It is important to understand the proposed real-world application of a model. Consulting domain experts could do this. They could help to fine-tune the scope of the model. It could also be of great importance to work with potential users of the model. They could provide useful insights that could help the model-building process be more objective. A more objective scope would mean fewer free parameters. The model would have a greater focus on its purpose. This would, in turn, keep the model as simple as possible.
+
+It is important to have the complexity of the model under control. The first step is understanding why we choose a specific model architecture for a given task. If we choose an architecture, it helps to consider whether there is a simpler architecture that could produce a similar result. We could also consider combining or reducing variables to limit complexity.
+
+#### Stress Tests
+
+Stress testing is the subjecting of machine learning models to various tests. These tests determine whether model performance is adequate under unfavorable or extreme conditions. Stress tests may help in the detection of underspecification. From this [paper](https://arxiv.org/pdf/2011.03395.pdf), we learn about three types of stress tests. Google researchers used them on their models, with various random seed values. A seed value can be thought of as a parameter that decides the sequence of generated pseudorandom numbers. [Pseudorandomness](https://en.wikipedia.org/wiki/Pseudorandomness) computes the extent to which a sequence of numbers appears to be statistically random, even though a repeatable process produces the numbers.
+
+The three types of stress tests include:
+
+**Stratified Performance Evaluations**. An [inductive bias](http://inductivebias.com/Blog/what-is-inductive-bias/) refers to a set of assumptions that a learner uses to make predictions given never-before encountered inputs. Stratified performance evaluations test whether a predictor encodes inductive biases that give similar performances across different dataset classes. These tests consider sub-groups of observations to identify inequalities.
+
+**Shifted Performance Evaluations**. Shifted performance evaluations evaluate whether the average performance generalizes when the test-set distribution is different from the training set distribution. This is done by spawning a made-up test distribution. This made-up distribution is created by deliberately causing perturbations on data. The perturbations are to keep track of the behavior of the model.
+
+**Contrastive Evaluations**. These tests manipulate an input observation to compute the significance of the effect on the output. These evaluations support localized analysis of specific inductive biases. They are performed on an example/observation as opposed to data distribution. The goal is to check whether modifying a given input causes the model's output to change in unanticipated ways.
```

Comment:
Does this come under explainable AI? If I can figure out which aspect contributes to the output, then can we explore this test for all models? What is the catch here?